apis/logging/v1/input_receiver_types.go

```diff
@@ -6,7 +6,17 @@ package v1
 //
 // The fields of this struct define the set of known Receiver types.
 type ReceiverSpec struct {
-	HTTP *HTTPReceiver `json:"http,omitempty"`
+	// Port the Receiver listens on.
+	// +required
```

Comment:
This will be an API regression.  We previously identified it as optional with a default value and thus the reason "GetPort" exists.  This would break any existing deployments on upgrade

Reply:
That's true and my first attempt respected older deployments, but @alanconway requested the API to be changed to use ReceiverTypeSpec the same way OutputTypeSpec was working. 

Existing deployments that use HTTPReceiver will fail whether we make the receiver port as required or not because currently the port is a child of HTTP and now it is a child of Receiver. Deployments that do not make use of the HTTPReceiver will not need any changes.

Reply:
@jlarriba sorry I wasn't clear - I do want a top level receiver spec to contain future different types of receiver, but I think the port should still be inside the HTTPReceiver. Unlike "Output" I don't think there's any "universal" config we can put a the top level for receivers.  As @syedriko points out below, it's possible we may have client-based or file scraping receivers in future, so I don't want to bake in the assumption that all receivers listen on a port. The existing release is explicitly preview so its OK if we break compatibility to enable future expansion.

Reply:
OK, understood. The port will be part of the specific Receiver and there will only be one common field for all Receivers: "Type".